lib/LruMemoize.js

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import LRU from 'lru-cache';
+
+export default class LruMemoize {
+  constructor(cacheOptions) {
+    this.cache = new LRU(cacheOptions);
+  }
+
+  async memoize({key, fn}) {
+    let promise = this.cache.get(key);
+    if(promise) {
+      return promise;
+    }
+
+    // cache miss
+    promise = fn();
```

Comment:
so the function passed in returns a promise? We probably should document that as an Async Function then.

Reply:
It is fine for `fn` to be a promise returning or non-promise returning function. In the non-promise case, the value returned from calling `fn()` will be converted to a resolved Promise on Line 23 (`result = await promise`).